src/basic/fs-util.c

```diff
@@ -75,46 +75,6 @@ int rmdir_parents(const char *path, const char *stop) {
         }
 }
 
-int rename_noreplace(int olddirfd, const char *oldpath, int newdirfd, const char *newpath) {
-        int r;
-
-        /* Try the ideal approach first */
-        if (renameat2(olddirfd, oldpath, newdirfd, newpath, RENAME_NOREPLACE) >= 0)
-                return 0;
-
-        /* renameat2() exists since Linux 3.15, btrfs and FAT added support for it later. If it is not implemented,
```

Comment:
hmm, iirc renameat2() support was added much later than 3.15 in some file systems. We shouldn't drop the compat kludge here unless we checked that all relevant file systems support it these days.